src/Microsoft.Azure.WebJobs.Host/Bindings/BindingProviders/AsyncCollectorBindingProvider.cs

```diff
@@ -206,17 +206,34 @@ public IEnumerable<BindingRule> GetRules()
             return rules;
         }
 
+        // Listed in precedence for providing via DefaultType.
+        private static readonly Type[] _defaultTypes = new Type[] { typeof(JObject), typeof(byte[]), typeof(string) };
+
         public Type GetDefaultType(Attribute attribute, FileAccess access, Type requestedType)
         {
             if (access == FileAccess.Write)
-            {              
-                var cm = (ConverterManager)this._converterManager;
-                var types = cm.GetPossibleSourceTypesFromDestination(attribute.GetType(), typeof(TType));
+            {
+                // TType is OpenType 
+                // TType is CloudMessage,   String --> CloudMessage
+
+                var ot = OpenType.FromType<TType>();
 
-                // search in precedence 
-                foreach (var target in new Type[] { typeof(JObject), typeof(byte[]), typeof(string) })
+                // Check for a direct match. 
```

Comment:
>// Check for a direct match.  [](start = 16, length = 29)

This section is the fix for [https://github.com/Azure/azure-webjobs-sdk/issues/1415](https://github.com/Azure/azure-webjobs-sdk/issues/1415)